src/codemodder/codemods/utils.py

```diff
@@ -104,6 +106,40 @@ def on_leave(self, original_node, updated_node):
         return updated_node
 
 
+class MetadataPreservingTransformer(
+    MatcherDecoratableTransformer, cst.MetadataDependent
+):
+    """
+    The CSTTransformer equivalent of ContextAwareVisitor. Will preserve metadata passed through a context. You should not chain more than one of these, otherwise metadata will not reflect the state of the tree.
+    """
+
+    def __init__(self, context: CodemodContext) -> None:
+        MetadataDependent.__init__(self)
+        MatcherDecoratableTransformer.__init__(self)
+        self.context = context
+        dependencies = self.get_inherited_dependencies()
+        if dependencies:
+            wrapper = self.context.wrapper
+            if wrapper is None:
+                # pylint: disable-next=broad-exception-raised
+                raise Exception(
```

Comment:
I would suggest `ValueError` since it's related to something being None, and remove the pylint disable

Reply:
This was actually copied directly from `libcst`'s `ContextAwareVisitor.__init__`. The only reason this class exists is because `ContextAwareTransformer` will handle metadata differently.

No reason we cannot improve as per your suggestion, just giving some context.